corehq/sql_db/util.py

```diff
@@ -101,3 +106,39 @@ def get_db_aliases_for_partitioned_query():
     else:
         db_names = ['default']
     return db_names
+
+
+def get_default_db_aliases():
+    return ['default']
+
+
+def get_default_and_partitioned_db_aliases():
+    return list(set(get_db_aliases_for_partitioned_query() + get_default_db_aliases()))
+
+
+def handle_connection_failure(get_db_aliases=get_default_db_aliases):
```

Comment:
@snopoke fyi i moved this decorator you had made into here, and made a couple changes: 1) i removed the retry ability because this is only used in the looping `GenericEnqueuingOperation` processes so it always gets retried anyway, and 2) i added the ability to cleanup from multiple dbs instead of just the default